sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -197,7 +197,7 @@ final class Decimal extends Ordered[Decimal] with Serializable {
     }
   }
 
-  override def toString: String = toBigDecimal.toString()
+  override def toString: String = toBigDecimal.bigDecimal.toPlainString()
```

Comment:
Yea, I was wondering `0E-7` vs `0.0000000`. Looks Postgres follows `0.0000000` way, right? 

cc @mengxr, it vaguely remember that I saw something you said about `0E-7` expression before. Please ignore me if I am wrongly remembering this.